juju/model.py

```diff
@@ -542,17 +542,26 @@ async def resolve(self, url, architecture, app_name=None, channel=None, series=N
         ch = Channel('latest', 'stable')
         if channel is not None:
             ch = Channel.parse(channel).normalize()
+
         origin = client.CharmOrigin(source="charm-hub",
                                     architecture=architecture,
                                     risk=ch.risk,
                                     track=ch.track)
-        charm_url, origin = await self.charm_resolver(url, origin)
+
+        charm_url, origin, supported_series = await self.charm_resolver(url, origin)
 
         if app_name is None:
             app_name = url.name
 
+        if series:
+            if series in supported_series:
+                origin.series = series
+                charm_url.series = series
+            else:
+                raise JujuError(f"Series {series} not supported for {str(url)}. Only {supported_series}")
```

Comment:
I'd change this to a slightly older syntax (because we still're supporting `Python 3.5` - until 3.0).

`"Series % not supported for %s. Only %s" % (series, str(url), supported_series)`

Reply:
We're using `"{}".format()` syntax in pretty much every other place.

Reply:
Oh that works with py35 too you're right. Though, I've been using the other syntax, I should probably go ahead and make a consistency update at some point